lib/matplotlib/lines.py

```diff
@@ -898,6 +925,8 @@ def set_color(self, color):
 
         ACCEPTS: any matplotlib color
         """
+        if color != self._color:
```

Comment:
need to watch out here, given that there can be multiple ways to represent the same color.